server/etcdserver/api/etcdhttp/health_test.go

```diff
@@ -39,13 +39,17 @@ import (
 
 type fakeHealthServer struct {
 	fakeServer
-	apiError      error
-	missingLeader bool
-	authStore     auth.AuthStore
+	serializableReadError error
+	linearizableReadError error
+	missingLeader         bool
+	authStore             auth.AuthStore
 }
 
-func (s *fakeHealthServer) Range(_ context.Context, _ *pb.RangeRequest) (*pb.RangeResponse, error) {
-	return nil, s.apiError
+func (s *fakeHealthServer) Range(_ context.Context, req *pb.RangeRequest) (*pb.RangeResponse, error) {
+	if req.Serializable || s.serializableReadError != nil {
```

Comment:
it should be like 

```
if req.Serializable {
   return nil, s.serializableReadError
}
```

Is it correct?

Reply:
My original thought is that if there is serializable error, linearizable read should fail as well. But that could be confusing. 
Changed for clarity.